graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/ast/expressions/AdditionExpression.java

```diff
@@ -92,6 +91,13 @@ public Object evaluateUnsafe(EvaluationContext context) {
             } else {
                 return new Duration(right, left);
             }
+        }else if(String.class.equals(type) ){
+            if(!isPlus){
+                return null;
+            }
+            final String left = (String) leftValue;
```

Comment:
Both this and the next line warn about `Casting 'leftValue' to 'String' may produce 'ClassCastException'`.
While technically the code cannot throw that exception (`leftValue` is a string because we checked and `rightValue` has to be a string, too, because of the parser/typechecker checks, we try to avoid warnings in code when possible.

My suggestion would be to rewrite this as `String.valueOf(leftValue)` to be future proof.